public/app/features/alerting/unified/rule-list/GrafanaGroupLoader.tsx

```diff
@@ -0,0 +1,144 @@
+import { useMemo } from 'react';
+
+import { Alert } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import { GrafanaRuleGroupIdentifier } from 'app/types/unified-alerting';
+import { GrafanaPromRuleDTO, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { alertRuleApi } from '../api/alertRuleApi';
+import { prometheusApi } from '../api/prometheusApi';
+import { GrafanaRulesSource } from '../utils/datasource';
+
+import { GrafanaRuleListItem } from './GrafanaRuleLoader';
+import { RuleOperationListItem } from './components/AlertRuleListItem';
+import { AlertRuleListItemLoader } from './components/AlertRuleListItemLoader';
+import { RuleOperation } from './components/RuleListIcon';
+
+const { useGetGrafanaRulerGroupQuery } = alertRuleApi;
+const { useGetGrafanaGroupsQuery } = prometheusApi;
+
+export interface GrafanaGroupLoaderProps {
+  groupIdentifier: GrafanaRuleGroupIdentifier;
+  namespaceName: string;
+  expectedRulesCount?: number;
+}
+
+export function GrafanaGroupLoader({
+  groupIdentifier,
+  namespaceName,
+  expectedRulesCount = 3,
```

Comment:
I would add a comment here and use a const for this 3. Something like 'we use this 3 for the default skeleton list of rules in case we don't know how many rules are in this group'..?